core/src/apps/cardano/sign_tx.py

```diff
@@ -61,18 +75,34 @@ async def sign_tx(
     return tx
 
 
+def _validate_network_info(network_id: int, protocol_magic: int) -> None:
+    is_mainnet_network_id = network_ids.is_mainnet(network_id)
+    is_mainnet_protocol_magic = protocol_magics.is_mainnet(protocol_magic)
+
+    if is_mainnet_network_id != is_mainnet_protocol_magic:
+        raise wire.ProcessError("Invalid network id/protocol magic combination!")
+
+
 def _validate_outputs(
-    keychain: seed.Keychain, outputs: List[CardanoTxOutputType], protocol_magic: int
+    keychain: seed.Keychain,
+    outputs: List[CardanoTxOutputType],
+    protocol_magic: int,
+    network_id: int,
 ) -> None:
+    if len(outputs) < 1:
```

Comment:
`if not outputs:`

Reply:
Done [here](https://github.com/trezor/trezor-firmware/pull/1112/commits/4cd90907cff79e58ae50a2d0fd699b6d6775c019).